runtime/src/server/middleware/get_page_handler.ts

```diff
@@ -237,7 +237,8 @@ export function get_page_handler(
 							host: req.headers.host,
 							path: req.path,
 							query: req.query,
-							params
+							params,
+							error
```

Comment:
It looks to me like this could also be a `string` here. If you look 10 lines down then you see how we standardize the `error` to an `Error` type. We should probably do that earlier and use the standardized error in both places

Should we also return `status` here as well?

Reply:
> It looks to me like this could also be a string here. 

True; I didn't see the conversion code. I think I'd prefer relaxing the type of `error` to `any`. It's consistent with how TypeScript typically types errors and there is generally [no guarantee that what you `catch` is of any particularly type](https://github.com/microsoft/TypeScript/issues/8677#issuecomment-220385124), so `Error | string` isn't correct.

The conversion is also a bit funky: `{ message: error }` is not an instance of `Error`.

I find only two places where a string is thrown: [on a 404](https://github.com/sveltejs/sapper/blob/master/runtime/src/server/middleware/get_page_handler.ts#L371) where the string is `Not found` and [on an error in preload](https://github.com/sveltejs/sapper/blob/master/runtime/src/server/middleware/get_page_handler.ts#L213) that gets converted into just the message. I'd propose changing them to `new Error('Not found')` and the actual caught error from preload rather than just the message.

This is technically a breaking change, but it's highly unlikely anyone depends on this behavior. 

Then we could also remove `|| new Error('Unknown error in preload function')` [in `handle_error`](https://github.com/sveltejs/sapper/blob/master/runtime/src/server/middleware/get_page_handler.ts#L43) that doesn't seem right, given that the function is called from other places than just `preload`. 

Should I go ahead with the changes described above?

> Should we also return `status` here as well?

It doesn't seem particularly useful to the layout, but on the other hand it does go together with the `error` object in a way. So we probably should. I'll add it.

(Though I think I would have preferred the status code as a property on a specialized error subclass, so we could throw e.g. `NotFoundError` and then `e.status == 404`, but that's a bigger change)

Reply:
I'd probably rather we continue to accept a `string`. It's both a bit more concise for many use cases and avoids making this a breaking change. I think that we can coalesce it to a single `Error` type later in the code as mentioned below.

Good catch on the fallback error message. It should probably be something like "Unknown error" instead of "Unknown error in preload function". Hopefully that error would never actually be shown - the developer would have to pass an empty string or similar which I would hope no one would do

Reply:
Ok, I've updated the message.